.github/workflows/main.yml

```diff
@@ -52,12 +52,14 @@ jobs:
 
     steps:
       - name: Check out the repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           persist-credentials: false
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4.3.0
```

Comment:
curious why pin to a patch version?  are we better off just pinning to a major one?

Reply:
If i recall this was initially kicked off https://github.com/dbt-labs/dbt-core/pull/6204 and might be to start bringing in 3.11 but being far back enough to not lose a older version of python? and to handle some deprecations. Making this change now because of new issues popping up in ci/cd will bring it into parity with the rest of bigquery/adapters https://github.com/dbt-labs/dbt-bigquery/pull/378

Reply:
cool thanks for the context!